weave/trace/vals.py

```diff
@@ -87,19 +87,30 @@ def make_mutation(
 
 
 class Traceable:
-    mutated_value: Any = None
-    ref: RefWithExtra
-    list_mutations: Optional[list] = None
+    ref: Optional[RefWithExtra]
     mutations: Optional[list[Mutation]] = None
     root: "Traceable"
+    parent: Optional["Traceable"] = None
     server: TraceServerInterface
+    _is_dirty: bool = False
+
+    def _mark_dirty(self) -> None:
+        """Recursively mark this object and its ancestors as dirty and removes their refs."""
+        self._is_dirty = True
+        self.ref = None
+        if self.parent is not self and self.parent is not None:
+            self.parent._mark_dirty()
 
     def add_mutation(
         self, path: tuple[str, ...], operation: MutationOperation, *args: Any
     ) -> None:
+        """Adds a mutation to the object.
+
+        TODO: This function is currently unused"""
```

Comment:
i'd elaborate to say: `we are tracking mutations for bookkeeping purposes but are not actually using them yet. In the future, we may choose to only send mutation updates over the wire` ... something that helps the reader understand

Reply:
I'll just remove commentary here since it's not part of this PR and will be part of the actual mutation-applying in a subsequent PR.  This function existed before this PR.